### PR TITLE
issue-4-delete-art

### PR DESCRIPTION
**Pull request by dennisliu9**
ajax-project_feature-4_delete_solution
ajax-project_feature-4_delete-mobile_solution

### CODE REVIEW

File: js/data.js [hunk 1]
```diff
@@ -36,7 +37,13 @@ var metadata = {
 };
 
 if (localStorage.getItem(localStorageKey) !== null) {
-  data = JSON.parse(localStorage.getItem(localStorageKey));
+  var referenceDataKeys = Object.keys(data);
+  var tmpData = JSON.parse(localStorage.getItem(localStorageKey));
+  if (JSON.stringify(referenceDataKeys) === JSON.stringify(Object.keys(tmpData))) {
```
Comment (thebearingedge) [inline js/data.js]: Using JSON.stringify this way works, but is not efficient and only works in certain situations. It's best to do a quick comparison.

```js
function equalArrays(first, second) {
  if (first.length !== second.length) return false;
  for (var i = 0; i < first.length; i++) {
    if (first[i] !== second[i]) return false;
  }
  return true;
}
```

File: js/main.js [hunk 2]
```diff
@@ -98,18 +107,28 @@ $bottomSheetHeader.addEventListener('click', function (event) {
...
     $bottomSheet.classList.remove('inner-scroll');
-    $bottomSheetCloseButton.classList.add('invisible');
+    $bottomSheetButtons.classList.add('invisible');
     $bottomSheetExpandButton.textContent = 'expand_less';
     $mainAppArea.classList.remove('no-scroll');
     $mainAppArea.classList.add('inner-scroll');
+    toggleDeleteMode(false);
+  } else if (event.target.tagName === 'SPAN' && ['delete_forever'].includes(event.target.textContent)) {
+    // turn on delete mode
+    clearInterval(deleteModeInfoBoxTimerId);
```
Comment (thebearingedge) [inline js/main.js]: I'm almost certain that clearInterval and clearTimeout do the same thing, but it's confusing to use clearInterval when your intent is to clear a timeout.

File: js/main.js [hunk 3]
```diff
@@ -496,32 +527,63 @@ function renderAllLiked() {
...
+      // Setting requestFullSize to false, images are VERY large
+      addImageToImg(data.viewingInDetail, $detailModalImage, false);
+      $detailModalContainer.classList.remove('hidden');
+    } else if (deleteMode === true) {
+      // Delete Mode on, show delete modal
+
+      // Put the selected image's data into data.deleting
+      for (var d = 0; d < data.likedObjects.length; d++) {
+        if (String(data.likedObjects[d].objectID) === String(event.target.getAttribute('objectId'))) {
+          data.deleting = data.likedObjects[d];
```
Comment (thebearingedge) [inline js/main.js:561]: When you find the matching id, include a `break` to stop further iteration through the array.

```js
        if (String(data.likedObjects[d].objectID) === String(event.target.getAttribute('objectId'))) {
          data.deleting = data.likedObjects[d];
          break;
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: js/data.js (hunk 1 above)
Comment (dennisliu9) [reply js/data.js]: Thanks, not sure why I didn't just do this from the start

File: js/main.js (hunk 2 above)
Comment (dennisliu9) [reply js/main.js]: That was my mistake, I was probably thinking of intervals when I was writing that...

File: js/main.js (hunk 3 above)
Comment (dennisliu9) [reply js/main.js:561]: I've added `break` to both the Liked search and the To Delete search, and I'll keep that in mind for similar loops in the future.

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**